src/user-guide/writing-workflows/external-triggers.rst

```diff
@@ -207,137 +451,127 @@ To see this, take a look at the job script for one of the downstream tasks:
 
-By setting the workflow-wide :cylc:conf:`flow.cylc[scheduling]sequential xtriggers`
-(defaults to ``False``) and/or keyword argument ``sequential`` to ``True``/``False`` in
-the xtrigger declaration:
+.. code-block:: cylc
 
-.. literalinclude:: ../../workflows/xtrigger/sequential/flow.cylc
-   :language: cylc
+   [scheduling]
+       initial cycle point = now
+       [[xtriggers]]
+           echo_1 = echo(succeeded=True, hello, 99, point=%(point)s, foo=10)
+       [[graph]]
+           PT1H = "@echo_1 => foo"
+   [runtime]
+       [[foo]]
+           script = "printenv | grep echo_1"
 
-When implementing a :ref:`custom xtrigger <Custom Trigger Functions>`, you can
-set the default for the ``sequential`` keyword argument in the xtrigger function
-definition itself:
+Run this with ``cylc play --no-detach <workflow>`` and watch your terminal to see
+the xtrigger calls. View the task job log with ``cylc cat-log <workflow_id>//1/foo``
+to confirm that the dependent task received the xtrigger results.
 
-.. code-block:: python
 
-   def my_xtrigger(my_in, my_out, sequential=True)
+xrandom
+^^^^^^^
+
+The toy ``xrandom`` function sleeps for a configurable amount of time (useful for
+testing the effect of a long-running trigger function - which should be avoided)
+and has a configurable random chance of success. The function signature is:
+
+.. automodule:: cylc.flow.xtriggers.xrandom
+   :members: xrandom, validate
+   :member-order: bysource
 
-Xtrigger declaration takes precedence over function, and function over workflow
-wide setting. So the above workflow definition would read:
+An example xrandom trigger workflow:
 
-- ``foo`` spawns out to the runahead limit.
-- ``FAM`` spawns only when ``@upstream`` is satisfied.
-- All associated xtriggers are checked and, as expected, their satisfaction
-  is a prerequisite to task readiness.
+.. literalinclude:: ../../workflows/xtrigger/xrandom/flow.cylc
+   :language: cylc
 
 
 .. _Custom Trigger Functions:
 
-Custom Trigger Functions
-------------------------
+Custom Xtrigger Functions
+-------------------------
 
-Trigger functions are just normal Python functions, with a few special
-properties:
+Xtrigger functions are Python functions with some special requirements.
 
-- They must:
 
-  - Be defined in a module with the same name as the function, unless
-    provided using the ``cylc.xtriggers`` entry point;
-  - be compatible with the same Python version that runs the scheduler
-    (see :ref:`Requirements` for the latest version specification).
-
-- They can be located either:
+Requirements
+^^^^^^^^^^^^
 
-  - In ``<workflow-dir>/lib/python/``;
-  - Anywhere in your ``$CYLC_PYTHONPATH``;
-  - Defined using the ``cylc.xtriggers`` entry point for an installed
-    package - see :ref:`developing.xtrigger.plugins`
+Xtrigger functions must be compatible with the Python version that runs the
+scheduler (see :ref:`Requirements` for the latest version specification).
```

Comment:
Worth adding a note about portability? 

Or adding a note to the document on [Portable workflows](https://github.com/cylc/cylc-doc/blob/master/src/workflow-design-guide/portable-workflows.rst) pointing out that just because you've deployed on Python 3.12 doesn't mean that Walruses and `f'{myvar=}` will work if someone else has Cylc deployed with Python 3.7?